In [4]:
import pandas as pd
data=pd.read_csv("/content/tips.csv")
data


total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

Find out how many male and female persons gave tips more than 7.50?


In [5]:
gen_tip=data[data['tip']>7.50]
gen_tip['sex'].value_counts()

sex
Male    3
Name: count, dtype: int64

Display the all the rows where customer size is 2 .

In [6]:
size_1=data[data['size']==2]
size_1

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
3         23.68  3.31    Male     No   Sun  Dinner     2
6          8.77  2.00    Male     No   Sun  Dinner     2
8         15.04  1.96    Male     No   Sun  Dinner     2
9         14.78  3.23    Male     No   Sun  Dinner     2
..          ...   ...     ...    ...   ...     ...   ...
237       32.83  1.17    Male    Yes   Sat  Dinner     2
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[156 rows x 7 columns]

where customer size iDisplay all rows in ascending order which is determined by total_bill ?


In [7]:
total_bill=data['total_bill'].sort_values()
total_bill

67      3.07
92      5.75
111     7.25
172     7.25
149     7.51
       ...  
182    45.35
156    48.17
59     48.27
212    48.33
170    50.81
Name: total_bill, Length: 244, dtype: float64

Explain the code: tips[‘day’].describe()
ans:- this code shows the the total number of count of the days

In [9]:
data['day'].describe()


count     244
unique      4
top       Sat
freq       87
Name: day, dtype: object

What will be the output. tips.info


In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


Display  records of 6 number row   to 243 number row using slicing processdure.

In [12]:
data[6:244]

total_bill   tip     sex smoker   day    time  size
6          8.77  2.00    Male     No   Sun  Dinner     2
7         26.88  3.12    Male     No   Sun  Dinner     4
8         15.04  1.96    Male     No   Sun  Dinner     2
9         14.78  3.23    Male     No   Sun  Dinner     2
10        10.27  1.71    Male     No   Sun  Dinner     2
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[238 rows x 7 columns]